### ISSUE TITLE
ExtendedResourceToleration adds tolerations even when the quantity of requested resources is "0"

### ISSUE THREAD

**Issue by hrk091** [kubernetes/kubernetes#125426]
### What happened?

ExtendedResourceToleration is useful for setting tolerations automatically to pods requesting extended resources like GPUs, but the tolerations are still given even if the quantity of extended resources is set to “0”.

### What did you expect to happen?

ExtendedResourceToleration admission should skip adding toleration if the quantity of extended resources is set to “0”.

### How can we reproduce it (as minimally and precisely as possible)?

You can reproduce it by creating the following pod:

```
❯ cat pod-with-extended-resource.yaml    
apiVersion: v1
kind: Pod
metadata:
  name: debug
spec:
  containers:
  - command:
    - /bin/sh
    - -c
    - sleep infinity
    image: busybox
    name: debug
    resources:
      limits:
        nvidia.com/gpu: "0"

❯ kubectl apply -f pod-with-extended-resource.yaml 
pod/debug created

❯ kubectl get po debug -o jsonpath='{.spec.tolerations[?(@.key == "nvidia.com/gpu")]}' | jq
{
  "effect": "NoSchedule",
  "key": "nvidia.com/gpu",
  "operator": "Exists"
}
```

### Anything else we need to know?

It seems that this issue can be resolved by adding an additional conditional check whether the quantity of the extended resources is over 0 here: 
https://github.com/kubernetes/kubernetes/blob/a7ca13ea29ba5b3c91fd293cdbaec8fb5b30cee2/plugin/pkg/admission/extendedresourcetoleration/admission.go#L72-L85

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.28.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.5
```

</details>

### Cloud provider

N/A

**Comment by Aaina26** [comment]
/sig scheduling
/area hw-accelerators
For reference see: https://github.com/kubernetes/kubernetes/pull/55839

**Comment by mauri870** [comment]
I agree that passing along the value "0" does not seem to have much use, but given that this is allowed we might have someone actually relying on this behavior in the wild. See [Hyrum's Law](https://www.hyrumslaw.com/)
